File: output/GPT4o-Mini/xCodeEval_4000/C++/Go/f8ab5a25466b4fa266c7c4e14aaf6602.go
package main

import (
	"fmt"
	"math/big"
	"strings"
)

func intToString(n int64) string {
	return fmt.Sprintf("%d", n)
}

func sum(a, b string) string {
	maxLen := len(a)
	if len(b) > maxLen {
		maxLen = len(b)
	}
	a = fmt.Sprintf("%0*s", maxLen, a)
	b = fmt.Sprintf("%0*s", maxLen, b)

	vect := make([]int64, maxLen)
	for i := maxLen - 1; i >= 0; i-- {
		vect[i] = int64(a[i]-'0') + int64(b[i]-'0')
	}

	for i := maxLen - 1; i >= 0; i-- {
		if vect[i] > 9 {
			if i == 0 {
				vect = append(vect, vect[i]/10)
				vect[i] %= 10
			} else {
				vect[i-1] += vect[i] / 10
				vect[i] %= 10
			}
		}
	}

	res := strings.Builder{}
	for _, v := range vect {
		res.WriteByte(byte(v + '0'))
	}
	return strings.TrimLeft(res.String(), "0")
}

func multiply(a, b string) string {
	vect := make([][]int64, len(a))
	for i := range vect {
		vect[i] = make([]int64, 0)
	}

	for i := len(a) - 1; i >= 0; i-- {
		for u := 0; u < len(a)-i-1; u++ {
			vect[len(a)-i-1] = append(vect[len(a)-i-1], 0)
		}
		for j := len(b) - 1; j >= 0; j-- {
			vect[len(a)-i-1] = append(vect[len(a)-i-1], int64(a[i]-'0')*int64(b[j]-'0')))
		}
		for u := 0; u < len(vect[len(a)-i-1]); u++ {
			if vect[len(a)-i-1][u] > 9 {
				if u == len(vect[len(a)-i-1])-1 {
					vect[len(a)-i-1] = append(vect[len(a)-i-1], vect[len(a)-i-1][u]/10)
					vect[len(a)-i-1][u] %= 10
				} else {
					vect[len(a)-i-1][u+1] += vect[len(a)-i-1][u] / 10
					vect[len(a)-i-1][u] %= 10
				}
			}
		}
	}

	res := make([]string, len(a))
	for i := 0; i < len(a); i++ {
		for p := len(vect[i]) - 1; p >= 0; p-- {
			res[i] += string(vect[i][p] + '0')
		}
	}

	ans := "0"
	for _, r := range res {
		ans = sum(ans, r)
	}
	return ans
}

func rec1(n int64) int64 {
	ans := int64(0)
	switch n {
	case 1:
		for i := 0; i < 10; i++ {
			if i != 0 {
				ans++
			}
		}
	case 2:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				if i != 0 && i != k {
					ans++
				}
			}
		}
	case 3:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					if i != 0 && i != k && i != c && k != c {
						ans++
					}
				}
			}
		}
	case 4:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					for d := 0; d < 10; d++ {
						if i != 0 && i != k && i != c && i != d && k != c && k != d && c != d {
							ans++
						}
					}
				}
			}
		}
	case 5:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					for d := 0; d < 10; d++ {
						for e := 0; e < 10; e++ {
							if i != 0 && i != k && i != c && i != d && i != e && k != c && k != d && k != e && c != d && c != e && d != e {
								ans++
							}
						}
					}
				}
			}
		}
	case 6:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					for d := 0; d < 10; d++ {
						for e := 0; e < 10; e++ {
							for f := 0; f < 10; f++ {
								if i != 0 && i != k && i != c && i != d && i != e && i != f && k != c && k != d && k != e && k != f && c != d && c != e && c != f && d != e && d != f && e != f {
									ans++
								}
							}
						}
					}
				}
			}
		}
	case 7:
		for i := 1; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					for d := 0; d < 10; d++ {
						for e := 0; e < 10; e++ {
							for f := 0; f < 10; f++ {
								for g := 0; g < 10; g++ {
									if i != k && i != c && i != d && i != e && i != f && i != g && k != c && k != d && k != e && k != f && k != g && c != d && c != e && c != f && c != g && d != e && d != f && d != g && e != f && e != g && f != g {
										ans++
									}
								}
							}
						}
					}
				}
			}
		}
	case 8:
		ans = 1632960
	case 9:
		ans = 3265920
	case 10:
		ans = 3265920
	}
	return ans
}

func rec(n int64) int64 {
	ans := int64(0)
	switch n {
	case 1:
		for i := 0; i < 10; i++ {
			ans++
		}
	case 2:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				if i != k {
					ans++
				}
			}
		}
	case 3:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					if i != k && i != c && k != c {
						ans++
					}
				}
			}
		}
	case 4:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					for d := 0; d < 10; d++ {
						if i != k && i != c && i != d && k != c && k != d && c != d {
							ans++
						}
					}
				}
			}
		}
	case 5:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					for d := 0; d < 10; d++ {
						for e := 0; e < 10; e++ {
							if i != k && i != c && i != d && i != e && k != c && k != d && k != e && c != d && c != e && d != e {
								ans++
							}
						}
					}
				}
			}
		}
	case 6:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					for d := 0; d < 10; d++ {
						for e := 0; e < 10; e++ {
							for f := 0; f < 10; f++ {
								if i != k && i != c && i != d && i != e && i != f && k != c && k != d && k != e && k != f && c != d && c != e && c != f && d != e && d != f && e != f {
									ans++
								}
							}
						}
					}
				}
			}
		}
	case 7:
		for i := 0; i < 10; i++ {
			for k := 0; k < 10; k++ {
				for c := 0; c < 10; c++ {
					for d := 0; d < 10; d++ {
						for e := 0; e < 10; e++ {
							for f := 0; f < 10; f++ {
								for g := 0; g < 10; g++ {
									if i != k && i != c && i != d && i != e && i != f && i != g && k != c && k != d && k != e && k != f && k != g && c != d && c != e && c != f && c != g && d != e && d != f && d != g && e != f && e != g && f != g {
										ans++
									}
								}
							}
						}
					}
				}
			}
		}
	case 8:
		ans = 1814400
	case 9:
		ans = 3628800
	case 10:
		ans = 3628800
	}
	return ans
}

func main() {
	var x string
	for {
		if _, err := fmt.Scan(&x); err != nil {
			break
		}

		ans := "1"
		vis := make([]int, 101)
		for i := range vis {
			vis[i] = 0
		}

		CHn := int64(0)
		F := int64(0)
		for i := 0; i < len(x); i++ {
			if x[i] >= 'A' && x[i] <= 'J' && i != 0 && vis[x[i]-'A'] == 0 {
				CHn++
				vis[x[i]-'A'] = 1
			} else if x[i] >= 'A' && x[i] <= 'J' && i == 0 && vis[x[i]-'A'] == 0 {
				F++
				vis[x[i]-'A'] = 1
			}
		}

		Zeros := int64(0)
		if x[0] == '?' {
			ans = "9"
		}
		for i := 1; i < len(x); i++ {
			if x[i] == '?' {
				Zeros++
			}
		}
		if F != 0 {
			ans = multiply(intToString(rec1(F+CHn)), ans)
		} else if CHn > 0 {
			ans = multiply(intToString(rec(CHn)), ans)
		}

		fmt.Print(ans)
		for i := int64(0); i < Zeros; i++ {
			fmt.Print("0")
		}
		fmt.Println()
	}
}

File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/C/f8ab5a25466b4fa266c7c4e14aaf6602.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* id0(long long n) {
    char* ans = (char*)malloc(21); // Enough for long long
    sprintf(ans, "%lld", n);
    return ans;
}

char* sum(char* a, char* b) {
    int len_a = strlen(a);
    int len_b = strlen(b);
    int max_len = len_a > len_b ? len_a : len_b;
    
    char* result = (char*)malloc(max_len + 2); // +1 for carry, +1 for null terminator
    int carry = 0, index = 0;

    for (int i = 0; i < max_len; i++) {
        int digit_a = (len_a - 1 - i >= 0) ? a[len_a - 1 - i] - '0' : 0;
        int digit_b = (len_b - 1 - i >= 0) ? b[len_b - 1 - i] - '0' : 0;
        int sum = digit_a + digit_b + carry;
        result[index++] = (sum % 10) + '0';
        carry = sum / 10;
    }
    if (carry) {
        result[index++] = carry + '0';
    }
    result[index] = '\0';

    // Reverse the result
    for (int i = 0; i < index / 2; i++) {
        char temp = result[i];
        result[i] = result[index - 1 - i];
        result[index - 1 - i] = temp;
    }
    return result;
}

char* multiply(char* a, char* b) {
    int len_a = strlen(a);
    int len_b = strlen(b);
    char** vect = (char**)malloc(len_a * sizeof(char*));

    for (int i = 0; i < len_a; i++) {
        vect[i] = (char*)calloc(len_b + len_a + 1, sizeof(char)); // Enough space for result
        for (int j = 0; j < len_a - i - 1; j++) {
            vect[i][j] = '0';
        }
        for (int j = len_b - 1; j >= 0; j--) {
            int product = (a[len_a - 1 - i] - '0') * (b[j] - '0');
            sprintf(vect[i] + strlen(vect[i]), "%d", product);
        }
    }

    char* ans = id0(0);
    for (int i = 0; i < len_a; i++) {
        char* temp = sum(ans, vect[i]);
        free(ans);
        ans = temp;
    }
    for (int i = 0; i < len_a; i++) {
        free(vect[i]);
    }
    free(vect);
    return ans;
}

long long rec1(long long n) {
    long long ans = 0;
    if (n == 1) {
        for (int i = 0; i < 10; i++) {
            if (i != 0) ans++;
        }
    } else if (n == 2) {
        for (int i = 0; i < 10; i++) {
            for (int k = 0; k < 10; k++) {
                if (i != 0 && i != k) ans++;
            }
        }
    } else if (n == 3) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    if (i != 0 && i != k && i != c && k != c) ans++;
    } else if (n == 4) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    for (int d = 0; d < 10; d++)
                        if (i != 0 && i != k && i != c && i != d && k != c && k != d && c != d) ans++;
    } else if (n == 5) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    for (int d = 0; d < 10; d++)
                        for (int e = 0; e < 10; e++)
                            if (i != 0 && i != k && i != c && i != d && i != e && k != c && k != d && k != e && c != d && c != e && d != e) ans++;
    } else if (n == 6) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    for (int d = 0; d < 10; d++)
                        for (int e = 0; e < 10; e++)
                            for (int f = 0; f < 10; f++)
                                if (i != 0 && i != k && i != c && i != d && i != e && i != f && k != c && k != d && k != e && k != f && c != d && c != e && c != f && d != e && d != f && e != f) ans++;
    } else if (n == 7) {
        for (int i = 1; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    for (int d = 0; d < 10; d++)
                        for (int e = 0; e < 10; e++)
                            for (int f = 0; f < 10; f++)
                                for (int g = 0; g < 10; g++)
                                    if (i != k && i != c && i != d && i != e && i != f && i != g && k != c && k != d && k != e && k != f && k != g && c != d && c != e && c != f && c != g && d != e && d != f && d != g && e != f && e != g && f != g) ans++;
    } else if (n == 8) {
        ans = 1632960;
    } else if (n == 9) {
        ans = 3265920;
    } else if (n == 10) {
        ans = 3265920;
    }
    return ans;
}

long long rec(long long n) {
    long long ans = 0;
    if (n == 1) {
        for (int i = 0; i < 10; i++) {
            ans++;
        }
    } else if (n == 2) {
        for (int i = 0; i < 10; i++) {
            for (int k = 0; k < 10; k++) {
                if (i != k) ans++;
            }
        }
    } else if (n == 3) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    if (i != k && i != c && k != c) ans++;
    } else if (n == 4) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    for (int d = 0; d < 10; d++)
                        if (i != k && i != c && i != d && k != c && k != d && c != d) ans++;
    } else if (n == 5) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    for (int d = 0; d < 10; d++)
                        for (int e = 0; e < 10; e++)
                            if (i != k && i != c && i != d && i != e && k != c && k != d && k != e && c != d && c != e && d != e) ans++;
    } else if (n == 6) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    for (int d = 0; d < 10; d++)
                        for (int e = 0; e < 10; e++)
                            for (int f = 0; f < 10; f++)
                                if (i != k && i != c && i != d && i != e && i != f && k != c && k != d && k != e && k != f && c != d && c != e && c != f && d != e && d != f && e != f) ans++;
    } else if (n == 7) {
        for (int i = 0; i < 10; i++)
            for (int k = 0; k < 10; k++)
                for (int c = 0; c < 10; c++)
                    for (int d = 0; d < 10; d++)
                        for (int e = 0; e < 10; e++)
                            for (int f = 0; f < 10; f++)
                                for (int g = 0; g < 10; g++)
                                    if (i != k && i != c && i != d && i != e && i != f && i != g && k != c && k != d && k != e && k != f && k != g && c != d && c != e && c != f && c != g && d != e && d != f && d != g && e != f && e != g && f != g) ans++;
    } else if (n == 8) {
        ans = 1814400;
    } else if (n == 9) {
        ans = 3628800;
    } else if (n == 10) {
        ans = 3628800;
    }
    return ans;
}

int main() {
    char x[101];
    while (scanf("%s", x) != EOF) {
        char* ans = id0(1);
        int vis[101] = {0};

        long long CHn = 0;
        long long F = 0;
        for (int i = 0; i < strlen(x); i++) {
            if ((x[i] >= 'A' && x[i] <= 'J') && i != 0 && !vis[x[i] - 'A']) {
                CHn++;
                vis[x[i] - 'A'] = 1;
            } else if ((x[i] >= 'A' && x[i] <= 'J') && i == 0 && !vis[x[i] - 'A']) {
                F++;
                vis[x[i] - 'A'] = 1;
            }
        }

        long long Zeros = 0;
        if (x[0] == '?') {
            ans = id0(9);
        }
        for (int i = 1; i < strlen(x); i++) {
            if (x[i] == '?') {
                Zeros++;
            }
        }
        if (F != 0) {
            char* temp = multiply(id0(rec1(F + CHn)), ans);
            free(ans);
            ans = temp;
        } else if (CHn > 0) {
            char* temp = multiply(id0(rec(CHn)), ans);
            free(ans);
            ans = temp;
        }

        printf("%s", ans);
        for (int i = 0; i < Zeros; i++) {
            printf("0");
        }
        printf("\n");
        free(ans);
    }
    return 0;
}
